### PR TITLE
Add Update Interactive tests

### PR DESCRIPTION
**Pull request by mzhubail**
re: #10796

Added two cases for update command interactive mode.

### CODE REVIEW

**Comment by mzhubail** [comment]
Regarding the failing windows test, It seems like CRLF is causing the issue, any pointers on fixing it?

Regarding the 8.4 test, I don't know what to do.

**Comment by Seldaek** [comment]
Sorry but I'm off atm and can't help much but 8.4 dont worry about it that's not your code, and windows same it's usually easy to fix.

**Comment by mzhubail** [comment]
Good to hear.

Just wanted to see that things are good overall, before I start working on other command.

File: tests/Composer/Test/Command/UpdateCommandTest.php [hunk 1]
```diff
@@ -12,7 +12,10 @@
 
 namespace Composer\Test\Command;
 
+use Composer\Package\Link;
+use Composer\Semver\Constraint\MatchAllConstraint;
 use Composer\Test\TestCase;
+use Generator;
```
Comment (stof) [inline tests/Composer/Test/Command/UpdateCommandTest.php:18]: AFAIK, composer follows the Symfony coding standard of not import classes of the global namespace with use statements. You should respect that rule.

File: tests/Composer/Test/Command/UpdateCommandTest.php [hunk 2]
```diff
@@ -135,4 +138,97 @@ public function testInteractiveModeThrowsIfNoPackageEntered(): void
...
+        $appTester->setInputs(array_merge(array_keys($packageNames), ['', 'yes']));
+        $appTester->run([
+            'command' => 'update', '--interactive' => true,
+            '--no-audit' => true,
+            '--dry-run' => true,
+        ]);
+
+        self::assertStringEndsWith(
+            trim($expected),
+            trim($appTester->getDisplay())
```
Comment (stof) [inline tests/Composer/Test/Command/UpdateCommandTest.php:204]: to solve the tests on Windows, use `$appTester->getDisplay(true)` to make it normalize line breaks in the output (as done in other tests)

**Pull request merged**